### ISSUE TITLE
GetMapName() in DHLib returns an undesired "/"

### ISSUE THREAD

**Issue by AndrewTheel** [DarklightGames/DarkestHour#644]
GetMapName() in class DHLib returns an undesired "/" at the beginning of the map name string.  Started after this change: https://github.com/DarklightGames/DarkestHour/commit/0bb2e32383d8264ddf5b2a696d04b28314815c40

> "map":"/DH-Barashka"

**Comment by handsm** [comment]
Hmmm .... odd one. Didn't work before my 'fix', doesn't work after it. Now it just doesn't work in a different situation.

Before I changed it was cropping a character off the map name. Now it is leaving an extra character. Having checked it out a little further, there's a difference in this function's output between different net modes. Before it worked in multiplayer but failed in single player. Now that's reversed.

The problem is the native GetLocalURL() function that's at the heart of our DHLib function. It returns a long, complex string - basically quite like a server start up line - and we only want to grab a small bit of that long string. And it returns different results in different net modes. I changed a 5 to a 4 (no. of characters trimmed), but it turns out there is no 'magic number' and we need some logic that is more flexible.

Did a bit more digging and there's another native function called GetURLMap(), which seems much better for us. But again, like GetLocalURL() it returns different results in different net modes. But this time it seems simple and predictable and so easy to handle in our GetMapName(). The only difference is that on a dedicated server it returns DH-MyMapName.rom and in single player or on a client we just get DH-MyMapName, without the .rom file extension.

Easy to handle that and the function can be reduced to a single line:

    return Repl(L.GetURLMap(), ".rom", "");`

Which is taking the GetURLMap() string and stripping any ".rom" from it.
